integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/LogicalPlanSerializer.java

```diff
@@ -58,7 +59,11 @@ public String serialize(LogicalPlan x) {
     try {
       return mapper.writeValueAsString(x);
     } catch (Throwable e) {
-      return "Unable to serialize: " + e.getMessage();
+      try {
+        return mapper.writeValueAsString("Unable to serialize: " + e.getMessage());
```

Comment:
🙄

Reply:
Nasty bug. More details can be found in this PR: https://github.com/OpenLineage/OpenLineage/pull/581
This should be out from this PR.